src/edges_analysis/analysis/beams.py

```diff
@@ -492,6 +495,40 @@ def at_freq(self, freq: np.ndarray) -> Beam:
             beam=interp_beam,
         )
 
+    def smoothed(
+        self,
+        model: mdl.Model = mdl.Polynomial(n_terms=12),
+    ) -> Beam:
+        """
+        Smoothes the beam within its same set of frequencies.
+
+        ----------
+
+        Returns
+        -------
+        beam
+            The Beam object smoothed over its same frequencies.
+        """
+        if len(self.frequency) < 3:
+            raise ValueError(
+                "Can't freq-interpolate beam that has fewer than three frequencies."
+            )
+
+        # Frequency smoothing
+        smooth_beam = np.zeros_like(self.beam)
+
+        for i, bm in enumerate(self.beam.T):
+            for j, b in enumerate(bm):
+                cached_model = model.at(x=self.frequency)
```

Comment:
you'll want to bring this line outside the loops to actually use it as a cache :-)

Reply:
@steven-murray done. please check :)